From 844d2b6f8fa36e98825cb55f3ea5af8f3d80b42d Mon Sep 17 00:00:00 2001
Message-Id: <844d2b6f8fa36e98825cb55f3ea5af8f3d80b42d.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Thu, 20 Nov 2014 13:36:02 +0530
Subject: [PATCH 64/75] MUST_REBASE [VPG]: adf/intel/vlv: attach planes to the
 respective pipe

In vlv, since the planes in a pipe are fixed, attach the planes
to the respective pipe during init.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I616614af31f088e97dee1b930ac2938cfd4a94da
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c |   26 +++++++++++++++++++++-
 drivers/video/adf/intel/core/vlv/vlv_pri_plane.c |    9 ++++++++
 drivers/video/adf/intel/core/vlv/vlv_sp_plane.c  |    9 ++++++++
 3 files changed, 43 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index 7da0d58..fd37ed5 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -82,7 +82,9 @@ static int vlv_initialize_disp(struct vlv_dc_config *vlv_config, int pipe,
 	struct vlv_pri_plane *pplane;
 	struct vlv_sp_plane *splane;
 	struct dsi_pipe *dsi_pipe;
-	int err;
+	struct intel_plane *intel_plane;
+	struct intel_pipe *intel_pipe;
+	int err = 0;
 
 	if (pipe > MAX_PIPES) {
 		dev_err(vlv_config->base.dev, "%s:invalid pipe", __func__);
@@ -138,11 +140,33 @@ static int vlv_initialize_disp(struct vlv_dc_config *vlv_config, int pipe,
 		intel_dc_config_add_pipe(&vlv_config->base,
 					 &dsi_pipe->base, pipe);
 		vlv_config->vdisp[pipe].type = type;
+		intel_pipe = &vlv_config->vdisp[pipe].pipe.dsi.base;
 	} else {
 		pr_err("ADF: %s: unsupported pipe type = %d\n", __func__, type);
 		err = -EINVAL;
+		return err;
 	}
 
+	/*
+	 * In this platform the plane and pipe are fixed and cannot be
+	 * moved across to a different pipe, hence set the attachment by
+	 * default over here
+	 */
+	/* Attach Primary plane to Pipe */
+	intel_plane = &vlv_config->vdisp[pipe].pplane.base;
+	if (intel_plane->ops->attach)
+		intel_plane->ops->attach(intel_plane, intel_pipe);
+
+	/* Attach Sprite1 to the pipe */
+	intel_plane = &vlv_config->vdisp[pipe].splane[0].base;
+	if (intel_plane->ops->attach)
+		intel_plane->ops->attach(intel_plane, intel_pipe);
+
+	/* Attach Sprite2 to the pipe */
+	intel_plane = &vlv_config->vdisp[pipe].splane[1].base;
+	if (intel_plane->ops->attach)
+		intel_plane->ops->attach(intel_plane, intel_pipe);
+
 	return err;
 }
 
diff --git a/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c b/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
index 82331aa..d7bebe0 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
@@ -265,6 +265,14 @@ static int vlv_pri_calculate(struct intel_plane *plane,
 	return 0;
 }
 
+static int vlv_pri_attach(struct intel_plane *plane, struct intel_pipe *pipe)
+{
+	/* attach the requested plane to pipe */
+	plane->pipe = pipe;
+
+	return 0;
+}
+
 static int vlv_pri_validate(struct intel_plane *plane,
 		struct intel_buffer *buf,
 		struct intel_plane_config *config)
@@ -367,6 +375,7 @@ static const struct intel_plane_ops vlv_pri_ops = {
 		.supported_formats = pri_supported_formats,
 		.n_supported_formats = ARRAY_SIZE(pri_supported_formats),
 	},
+	.attach = vlv_pri_attach,
 	.validate = vlv_pri_validate,
 	.flip = vlv_pri_flip,
 	.enable = vlv_pri_enable,
diff --git a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
index ecbc732..0de2191 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
@@ -367,6 +367,14 @@ static int vlv_sp_calculate(struct intel_plane *planeptr,
 	return 0;
 }
 
+static int vlv_sp_attach(struct intel_plane *plane, struct intel_pipe *pipe)
+{
+	/* attach the requested plane to pipe */
+	plane->pipe = pipe;
+
+	return 0;
+}
+
 static int vlv_sp_validate(struct intel_plane *plane, struct intel_buffer *buf,
 		struct intel_plane_config *config)
 {
@@ -585,6 +593,7 @@ static const struct intel_plane_ops vlv_sp_ops = {
 		.supported_formats = sprite_supported_formats,
 		.n_supported_formats = ARRAY_SIZE(sprite_supported_formats),
 	},
+	.attach = vlv_sp_attach,
 	.validate = vlv_sp_validate,
 	.flip = vlv_sp_flip,
 	.enable = vlv_sp_enable,
-- 
1.7.9.5

